scss/_patterns_form-validation.scss

```diff
@@ -18,261 +18,111 @@
-        @include vf-form-validation-input-caution-dark-theme;
+      &:hover {
+        background-color: $colors--theme--background-caution-hover;
       }
 
-      .is-error & {
-        @include vf-form-validation-input-error-dark-theme;
+      &:focus {
+        background-color: $colors--theme--background-caution-active;
       }
     }
 
     .p-form-validation__message {
-      .is-success & {
-        @include vf-form-validation-message-success-dark-theme;
-      }
-
-      .is-caution & {
-        @include vf-form-validation-message-caution-dark-theme;
+      @include vf-icon-warning($color: $color-caution);
+      color: $colors--theme--border-caution;
+
+      // FIXME: Temporary workaround for unsupported theme colors in icons
+      // stylelint-disable-next-line selector-max-type
+      body.is-dark &,
+      &.is-dark {
+        @include vf-icon-warning($color: map-get($colors-dark-theme--tinted-borders, caution), $color-symbol: $colors--dark-theme--background-default);
       }
 
-      .is-error & {
-        @include vf-form-validation-message-error-dark-theme;
+      // FIXME: Temporary workaround for unsupported theme colors in icons
+      // stylelint-disable-next-line selector-max-type
+      &.is-light,
+      &.is-paper,
+      body.is-light &,
+      body.is-paper & {
+        @include vf-icon-warning($color: $color-caution);
       }
     }
-  } @else {
+  }
+
+  .is-error {
     .p-form-validation__input {
-      .is-success & {
-        @include vf-form-validation-input-success-light-theme;
-      }
+      background-color: $colors--theme--background-negative-default;
+      border-bottom-color: $colors--theme--border-negative;
 
-      .is-caution & {
-        @include vf-form-validation-input-caution-light-theme;
+      &:hover {
+        background-color: $colors--theme--background-negative-hover;
       }
 
-      .is-error & {
-        @include vf-form-validation-input-error-light-theme;
+      &:focus {
+        background-color: $colors--theme--background-negative-active;
       }
     }
 
     .p-form-validation__message {
-      .is-success & {
-        @include vf-form-validation-message-success-light-theme;
+      @include vf-icon-error($color: $color-negative);
+      color: $colors--theme--border-negative;
+
+      // FIXME: Temporary workaround for unsupported theme colors in icons
+      // stylelint-disable-next-line selector-max-type
+      body.is-dark &,
+      &.is-dark {
+        @include vf-icon-error($color: map-get($colors-dark-theme--tinted-borders, negative), $color-symbol: $colors--dark-theme--background-default);
       }
 
-      .is-caution & {
-        @include vf-form-validation-message-caution-light-theme;
+      // FIXME: Temporary workaround for unsupported theme colors in icons
+      // stylelint-disable-next-line selector-max-type
+      &.is-light,
+      &.is-paper,
+      body.is-light &,
+      body.is-paper & {
+        @include vf-icon-error($color: $color-negative);
```

Comment:
Same here. While it may be the same value I think, for clarity use the tinder border color from the theme variables.